sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -33,8 +33,8 @@ import org.apache.spark.util.collection.BitSet
  * Performs a sort merge join of two child relations.
  */
 case class SortMergeJoinExec(
-    leftKeys: Seq[Expression],
-    rightKeys: Seq[Expression],
+    var leftKeys: Seq[Expression],
```

Comment:
@hvanhovell : ping !!! As expected, that doesn't work in all cases and few unit test are failing. I would go back to my original version if you have don't any other idea(s).